sdk/textanalytics/Azure.AI.TextAnalytics/src/GeneratorStubs/Resolutions.cs

```diff
@@ -0,0 +1,582 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.AI.TextAnalytics.Models;
+using Azure.Core;
+
+namespace Azure.AI.TextAnalytics
+{
+#pragma warning disable SA1402 // File may only contain a single type
+
+    #region Primitives
+
+    [CodeGenModel("AgeUnit")]
+    public readonly partial struct AgeUnit { }
+
+    [CodeGenModel("AreaUnit")]
+    public readonly partial struct AreaUnit { }
+
+    [CodeGenModel("DateTimeSubKind")]
+    public readonly partial struct DateTimeSubKind { }
+
+    [CodeGenModel("InformationUnit")]
+    public readonly partial struct InformationUnit { }
+
+    [CodeGenModel("LengthUnit")]
+    public readonly partial struct LengthUnit { }
+
+    [CodeGenModel("NumberKind")]
+    public readonly partial struct NumberKind { }
+
+    [CodeGenModel("RangeKind")]
+    public readonly partial struct RangeKind { }
+
+    [CodeGenModel("RelativeTo")]
+    public readonly partial struct RelativeTo { }
+
+    [CodeGenModel("SpeedUnit")]
+    public readonly partial struct SpeedUnit { }
+
+    [CodeGenModel("TemperatureUnit")]
+    public readonly partial struct TemperatureUnit { }
+
+    [CodeGenModel("TemporalModifier")]
+    public readonly partial struct TemporalModifier { }
+
+    [CodeGenModel("VolumeUnit")]
+    public readonly partial struct VolumeUnit { }
+
+    [CodeGenModel("WeightUnit")]
+    public readonly partial struct WeightUnit { }
+
+    #endregion
+
+    #region Resolutions
+
+    [CodeGenModel("AgeResolution")]
```

Comment:
@heaths : I'm not a fan of how I'm exposing these Resolution types, but I couldn't figure out how to compose an autorest transform that would turn them into public read-only types (i.e., types with internal constructors, and only getters with no setters). Do you know what the trick might be? Or, perhaps, is the current approach acceptable?

Reply:
By default, if they are only used in responses they should already be read-only. Is that not what you're seeing?

See https://github.com/Azure/azure-sdk-for-net/search?q=x-accessibility&type=code for some examples of how to add the `x-accessibility` via transform. Your `where` clause could be something like `$.definitions.*` and then your transform could check against a static list of which models you want to make public, for example.

Reply:
I talked to Michael and ended up filing a related bug: https://github.com/Azure/autorest.csharp/issues/2823. I will keep it like this for now. Since it's only a handful of simple classes, I don't think it's problematic.